x-pack/plugins/enterprise_search/server/collectors/connectors/telemetry.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient, Logger } from '@kbn/core/server';
+
+import { CONNECTORS_INDEX } from '@kbn/search-connectors';
+import { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+
+interface Telemetry {
+  native: {
+    total: number;
+  };
+  clients: {
+    total: number;
+  };
+}
+
+/**
+ * Register the telemetry collector
+ */
+
+export const registerTelemetryUsageCollector = (
+  usageCollection: UsageCollectionSetup,
+  client: IClusterClient,
+  log: Logger
+) => {
+  const telemetryUsageCollector = usageCollection.makeUsageCollector<Telemetry>({
+    type: 'connectors',
+    fetch: async () => fetchTelemetryMetrics(client, log),
+    isReady: () => true,
+    schema: {
+      native: {
+        total: { type: 'long' },
+      },
+      clients: {
+        total: { type: 'long' },
+      },
+    },
+  });
+  usageCollection.registerCollector(telemetryUsageCollector);
+};
+
+/**
+ * Fetch the aggregated telemetry metrics
+ */
+
+export const fetchTelemetryMetrics = async (
+  client: IClusterClient,
+  log: Logger
+): Promise<Telemetry> => {
+  const defaultTelemetryMetrics: Telemetry = {
+    native: {
+      total: 0,
+    },
+    clients: {
+      total: 0,
+    },
+  };
+
+  try {
+    const nativeCountResponse = await client.asInternalUser.count({
+      index: CONNECTORS_INDEX,
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                is_native: true,
+              },
+            },
+          ],
+          must_not: [
+            {
+              term: {
+                service_type: {
+                  value: 'elastic-crawler',
+                },
+              },
+            },
+          ],
+        },
+      },
+    });
+
+    const clientsCountResponse = await client.asInternalUser.count({
+      index: CONNECTORS_INDEX,
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                is_native: false,
+              },
+            },
+          ],
+        },
+      },
+    });
+
+    return {
+      native: {
+        total: nativeCountResponse.count,
+      },
+      clients: {
+        total: clientsCountResponse.count,
+      },
+    } as Telemetry;
+  } catch (error) {
+    log.error(`Failed to retrieve telemetry data: ${error}`);
+    return defaultTelemetryMetrics;
```

Comment:
Are you sure you want to be reporting `0` counts on failure? maybe just letting the retrieval fail and the telemetry collector to handle it and not report anything is better

Reply:
Interested to know what will happen if this error propagates? Will it fail the whole telemetry collection (I guess not), or it just ignore this collector?

Reply:
correct it'll just ignore this collector, if you wish you can throw an error manually and test it by opening the telemetry flyout.

Reply:
I would actually prefer to return a 0 count, at least on an error caused by the indices not existing. That is going to be the most likely source of this error, but that's a valid state and means that the count is 0.

Additionally, I don't think we'd want to create a log line fore errors caused these indices not existing. That is going to be a common and perfectly valid situation and logging that as an error is just going to confuse users.

Reply:
Hi @sphilipse , I think it's better to raise errors and let the telemetry collector handle it.

In the later release, we want to wrap the telemetry collection into an API, and we will not be able to provide a reasonable default for the full telemetry (to be finalized in 8.13)